common/src/main/java/org/apache/drill/common/config/DrillConfig.java

```diff
@@ -164,6 +164,15 @@ public static DrillConfig create(String overrideFileResourcePathname, boolean en
   }
 
   /**
+   * Creates a drill configuration using the provided config file.
```

Comment:
What is this for? Is the caller responsible for loading the defaults? If this is primarily for unit testing, then the comment should say so and describe how to use this. (The config, for example, must provide all required properties; nothing will be picked up from defaults.)